benchmarks/benchmarks/signal_filtering.py

```diff
@@ -82,3 +84,54 @@ def setup(self, n_samples, order):
 
     def time_sosfilt_basic(self, n_samples, order):
         sosfilt(self.sos, self.y)
+
+
+class MedFilt2D(Benchmark):
+    def setup_cache(self):
+        np.random.seed(8176)
+        data = np.random.randn(2000, 4000)
+        expected = medfilt2d(data, 5)
+        np.save("medfilt2d_data.npy", data)
+        np.save("medfilt2d_expected.npy", expected)
+
+    def setup(self):
+        self.data = np.load("medfilt2d_data.npy")
+        self.expected = np.load("medfilt2d_expected.npy")
+
+    def teardown(self):
+        assert_allclose(self.output, self.expected)
+
+    def _multithreaded(self):
+        # Lets take four overlapping chunks over the last axis. First pair of
+        # values is range to select from input, second pair range of the result
+        # to use, last pair the range of the output array to store it in.
+        chunks = (
+            (0, 1002, None, -2, 0, 1000),
+            (998, 2002, 2, -2, 1000, 2000),
+            (1998, 3002, 2, -2, 2000, 3000),
+            (2998, 4000, 2, None, 3000, 4000)
+        )
+
+        self.output = np.empty(self.data.shape, dtype=self.data.dtype)
```

Comment:
`self.output` should be `output`. You don't want to save any class attributes within a benchmark.